CIP-XXXX/standard.cddl

```diff
@@ -0,0 +1,84 @@
+; Oracle Datum
+
+; The oracle datum is composed of a generic, shared and extended data.
+;   - With key 0: The generic data will be a list of different feed values.
+;   - With key 1: The shared data could not be part of the datum, but if it's
+;     present then carries the information about the timestamp and expiry of
+;     all the feeds.
+;   - With key 2: The extended data could not be part of the datum, but if it's
+;     present then carries some external information about the provider of the
+;     feeds.
+oracle_datum = #6.121(oracle_datum_list)
+oracle_datum_list =
+  [ ?  shared_data	
+  , 1* generic_data
+  , ?  extended_data
+  ]
+
+; At the moment we have two types of generic data:
+;   - price_data, for the exchange rate between two assets.
+;   - kyc_data, for “Know Your Customer” related data
+generic_data = price_data / kyc_data
+
+; A price_data entry normally defines the “Currency Pair” being priced.
+;  A “Currency Pair” is defined as “the first listed currency or base currency is
+;  bought while the second listed currency in a currency pair or quote currency is
+;  sold.” Each currency is defined separately in the entry, and referred to as the
+;  “base” and “quote” currency.
+price_data = #6.123([price_map])
+price_map =
+  { ? 0 : price        ; how many quote currency is received per base currency spent
+  , ? 1 : posixtime    ; unix timestamp related to when the price data was created
+  , ? 2 : posixtime    ; unix timestamp related to when the price data is expired
+  , ? 3 : precision    ; how many decimals of precision price is specified at
+  , ? 4 : asset_id     ; id of base
+  , ? 5 : asset_id     ; id of quote
+  , ? 6 : asset_symbol ; symbol of base
+  , ? 7 : asset_symbol ; symbol of quote
+  , ? 8 : asset_name   ; name of base
+  , ? 9 : asset_name   ; name of quote
+  }
+
+precision    = uint
+price        = uint
+asset_id     = uint
+asset_symbol = tstr
+asset_name   = tstr
+
+; In order:	
+; - Timestamp of the creation of all the generic data.
+; - Expiry of the creation of that data.
+shared_data = #6.121(shared_map)
+shared_map =
+  { ? 0: price_data
+  , ? 1: kyc_data
+  }
+
+; The extended data carries provider-specific information, and may
+;  be individually customized by oracle providers to provide data
+;  they deem relevant. It's thus not entirely defined by spec,
+;  but indices below 100 are "reserved" by the spec.
+extended_data = #6.122(extended_map)
+extended_map = 
+  { ? 0: oracle_provider_id
```

Comment:
If there is not a registry of which ids represent which providers, but dApp consumers are "fine to whitelist the providers", how would the dApp consumers know which ids represent which providers, in such a way that collisions aren't happening, without a registry?

I think you are reading too much "control" into a registry, it's not for a registry to accept and deny, it's for a registry to keep obvious spam out and prevent collisions.

Reply:
Suppose that there is sport betting protocol and an oracle provider for horse-racing results. The sport betting protocol wants to whitelist the oracle provider.

Under your proposal, the oracle provider would have to apply to the Cardano-wide oracle registry to obtain an integer ID, so that the sport betting protocol could whitelist that oracle provider using its integer ID.

---

Alternatively, if the oracle were instead identified by either an oracle signing key or minting policy ID, then the sport betting protocol could whitelist it simply by referring to that signing key or minting policy ID, without waiting for that oracle to be added to some registry.

A registry could still be useful in that case for dApps to discover potential oracle providers that it could whitelist, but dApps could also whitelist any oracle providers outside the registry if they so choose.

Reply:
If this registry was decentralized to an on-chain registry, as I touched on in a previous comment, this issue could perhaps be avoided. The sports betting protocol could then simply reserve their ID in advance, go ahead and implement without asking for permission while at the same time dealing with a pull-request to the standard

That way, if there was a stall in the CIP process the providers and any consumers could still go ahead and use their ID with full confidence that it will be 'theirs'. The standards process might still leave the final standardization for that ID being different than what the premature implementations used, but that's just a problem inherent to any consensus-driven standards process and not really related to the ID registry.